pinot-core/src/main/java/org/apache/pinot/core/operator/blocks/results/AggregationResultsBlock.java

```diff
@@ -45,10 +45,13 @@
 public class AggregationResultsBlock extends BaseResultsBlock {
   private final AggregationFunction[] _aggregationFunctions;
   private final List<Object> _results;
+  private final QueryContext _queryContext;
 
-  public AggregationResultsBlock(AggregationFunction[] aggregationFunctions, List<Object> results) {
+  public AggregationResultsBlock(AggregationFunction[] aggregationFunctions, List<Object> results,
+      QueryContext queryContext) {
     _aggregationFunctions = aggregationFunctions;
     _results = results;
+    _queryContext = queryContext;
```

Comment:
Does it make sense to put queryContext into BaseResultsBlock with a constructor and use super here?

Reply:
I considered that, but it doesn't apply to metadata and exception block, so I kept it only in data blocks